packages/ember-data/lib/system/relationships/belongs-to.js

```diff
@@ -86,9 +91,16 @@ function belongsTo(type, options) {
       } else {
         this._relationships[key].setRecord(value);
       }
+
+      return this._relationships[key].getRecord();
     }
+  };
 
-    return this._relationships[key].getRecord();
+  return Ember.computed(supportsComputedGetterSetter ? computedConfig : function(key, value) {
+    if (arguments.length > 1) {
+      return computedConfig.set.call(this, key, value);
+    }
+    return computedConfig.get.call(this, key);
```

Comment:
I considered using https://github.com/rwjblue/ember-new-computed/blob/master/addon/index.js from @rwjblue but didn't know if it was worth it since we only have settable CPs in these two places in ED. If we use a helper like the one linked we should probably make sure that we use it everywhere within ED.